### PR TITLE
@truffle/hdwallet-provider added typescript declaration file

### PR DESCRIPTION
**Pull request by MrRefactoring**

### CODE REVIEW

File: packages/hdwallet-provider/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import {
+  JSONRPCRequestPayload,
+  JSONRPCResponsePayload
+} from 'ethereum-protocol';
```
Comment (gnidan) [inline packages/hdwallet-provider/index.d.ts]: Looks like this is missing from the deps. (Wait, where does this sort of thing go... `dependencies` or `devDependencies`?)

**Comment by MrRefactoring** [comment]
> Thanks for this @MrRefactoring, that's awesome! Just the one thing about the missing dep.
> 
> Alternatively... how hard would it be to port this to TS? 😈

This would be very easy to port to typescript. If you are ready to switch to typescript, then I am ready to help

File: packages/hdwallet-provider/index.d.ts (hunk 1 above)
Comment (MrRefactoring) [reply packages/hdwallet-provider/index.d.ts]: `web3-provider-engine` use an `ethereum-protocol`. Better add this to `dependencies`?

Comment (MrRefactoring) [reply packages/hdwallet-provider/index.d.ts]: I also noticed that you did not add `debug` to `dependcies`

Comment (gnidan) [reply packages/hdwallet-provider/index.d.ts]: Yep, best to add all dependencies directly! Good catch on `debug`, that should be there too!

**Comment by gnidan** [comment]
> This would be very easy to port to typescript. If you are ready to switch to typescript, then I am ready to help

Definitely! We've been converting packages to TS here and there, incrementally, so if that's something you have availability for, we'd definitely welcome and appreciate it!

**Comment by gnidan** [comment]
Also if you pull upstream from `develop` I think I fixed the failing Slack notifier build so you won't have to see a red ❌

**Comment by MrRefactoring** [comment]
Migrated `@truffle/hdwallet-provider` to typescript in PR #2412

**Pull request closed without merging**